pandas/tests/dtypes/cast/test_infer_dtype.py

```diff
@@ -141,12 +141,29 @@ def test_infer_dtype_from_scalar_errors():
 
 
 @pytest.mark.parametrize(
-    "arr, expected, pandas_dtype",
+    "value, expected, pandas_dtype",
     [
         ("foo", np.object_, False),
         (b"foo", np.object_, False),
-        (1, np.int_, False),
+        (1, np.int64, False),
         (1.5, np.float_, False),
+        (np.datetime64("2016-01-01"), np.dtype("M8[ns]"), False),
+        (Timestamp("20160101"), np.dtype("M8[ns]"), False),
+        (Timestamp("20160101", tz="UTC"), np.object_, False),
+        (Timestamp("20160101", tz="UTC"), "datetime64[ns, UTC]", True),
```

Comment:
should we be return a DatetimeTZDtype object?

Reply:
we do

Reply:
> "datetime64[ns, UTC]" is a string, aren't we *specifically* return a DatetimeTZDtype object?

Reply:
yes, but this test uses is_dtype_equal to compare the result with expected

Reply:
ok (we did merge a similar patch for categorical so ok)